### PR TITLE
Update code to use the new network-path aware API of quiche

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Quiche changed its API to be network path aware and so be able to handle connection migration in the future.

Modifications:

- Adjust code to use the new API

Result:

Be able to use the new network-path aware API of quiche

### CODE REVIEW

File: src/main/c/netty_quic_quiche.c [hunk 1]
```diff
@@ -472,10 +575,71 @@ static jlong netty_buffer_memory_address(JNIEnv* env, jclass clazz, jobject buff
...
+  { "in6AddressOffsetofS6Addr", "()I", (void *) netty_quiche_in6AddressOffsetofS6Addr },
+  { "sizeofSockaddrStorage", "()I", (void *) netty_quiche_sizeofSockaddrStorage },
+  { "sizeofSizeT", "()I", (void *) netty_quiche_sizeofSizeT },
+  { "sizeofSocklenT", "()I", (void *) netty_quiche_sizeofSocklenT },
+  { "quicheRecvInfoOffsetofFrom", "()I", (void *) netty_quicheRecvInfoOffsetofFrom },
+  { "quicheRecvInfoOffsetofFromLen", "()I", (void *) netty_quicheRecvInfoOffsetofFromLen },
+  { "sizeofQuicheRecvInfo", "()I", (void *) netty_sizeofQuicheRecvInfo },
+  { "quicheSendInfoOffsetofTo", "()I", (void *) netty_quicheSendInfoOffsetofTo },
+  { "quicheSendInfoOffsetofToLen", "()I", (void *) netty_quicheSendInfoOffsetofToLen },
+  { "sizeofQuicheSendInfo", "()I", (void *) netty_sizeofQuicheSendInfo },
```
Comment (normanmaurer) [inline src/main/c/netty_quic_quiche.c:641]: While this looks like a lot of JNI it actually is mostly just wiring things up

File: src/main/java/io/netty/incubator/codec/quic/QuicheQuicCodec.java [hunk 2]
```diff
@@ -51,6 +53,7 @@
 
     protected final QuicheConfig config;
     protected final int localConnIdLength;
+    protected ByteBuf sockaddrMemory;
```
Comment (NiteshKant) [inline src/main/java/io/netty/incubator/codec/quic/QuicheQuicCodec.java:56]: Can we add some documentation on what is this buffer expected to contain?

File: src/main/java/io/netty/incubator/codec/quic/QuicheQuicConnection.java [hunk 3]
```diff
@@ -15,19 +15,28 @@
...
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
 import java.util.function.Supplier;
 
 final class QuicheQuicConnection {
+    private static final int TOTAL_RECV_INFO_SIZE = Quiche.SIZEOF_QUICHE_RECV_INFO + Quiche.SIZEOF_SOCKADDR_STORAGE;
+    private static final int QUICHE_SEND_INFOS_OFFSET = 2 * TOTAL_RECV_INFO_SIZE;
     private final ReferenceCounted refCnt;
     private final QuicheQuicSslEngine engine;
+    private final ByteBuf infoBuffer;
```
Comment (NiteshKant) [inline src/main/java/io/netty/incubator/codec/quic/QuicheQuicConnection.java:30]: Can we add some documentation on what is this buffer expected to contain?

File: src/main/java/io/netty/incubator/codec/quic/QuicheRecvInfo.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import io.netty.util.internal.PlatformDependent;
+
+import java.net.InetSocketAddress;
+
+final class QuicheRecvInfo {
+
+    private QuicheRecvInfo() { }
+
+    static void write(long memory, InetSocketAddress address) {
```
Comment (NiteshKant) [inline src/main/java/io/netty/incubator/codec/quic/QuicheRecvInfo.java:26]: Any comment or link to documentation of the logic behind this and sendInfo will help 😄

**Review by NiteshKant — APPROVED** [review]
It will take me some time to be productive in reviews in this repo but thanks for tagging me here 😄 

Left a few comments where some documentation may help or it may be just me being a noob here.

**Pull request merged**